src/esp/gfx/DrawableGroup.cpp

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+#include "DrawableGroup.h"
+
+namespace esp {
+namespace gfx {
+
+bool DrawableGroup::hasDrawable(uint64_t id) {
+  return (idToDrawable_.find(id) != idToDrawable_.end());
+}
+
+Drawable* DrawableGroup::getDrawable(uint64_t id) {
```

Comment:
```suggestion
Drawable const* DrawableGroup::getDrawable(uint64_t id) const {
```
How about this?  So that the address being pointed to isn't changed by the consumer accidentally, and the getter itself can be const too, right?

Reply:
Once we get the drawable object, we will not just simply query its data, but make changes to the drawable object. For example, in the coming diff, I will set this drawable as "selected" (a boolean member variable to be introduced to Drawable class) if it is picked by the mouse.

Reply:
Does that mean you may wish for the return value to point to another location in memory (i.e. like with a new assignment)?  <type> const* will just make sure the pointer always points to the same location, but what it is pointing at can change.  :)

Reply:
-typename- const * just means the memory address being pointed to by the pointer won't change.  The data at that memory can change.  Was thinking that we might not want folks to be able to reassign things externally

Reply:
Oh, now I understand what you really would like to do, which is a good point.

However, `typename const*` is the same as `const typename*`, and that means the object (which has type `typename`) is constant and you cannot change the value within. Certainly this is not the case we would like to have here.

So let me guess, you were trying to say, `typename* const`, which means a constant pointer to a (mutable) object. But I do not think we can use `typename* const` to the return value of a function because the `const` is totally redundant in this case. It is meaningless to add a type qualifier (const) on the return type itself (a pointer). It is ONLY meaningful to add `const` to the object the raw pointer is pointing to.

So the correct way to do what you really suggested is in the viewer:
we will have something like:
`esp::gfx::Drawable *const object = ...`

Reply:
Thanks for correcting me, Today has been a long day :).  I was thinking of the case where folks call things with auto&, and thinking that perhaps that could be dangerous.